### ISSUE TITLE
Windows PATH variables parsed with no '\'

### ISSUE THREAD

**Issue by okuma10** [nickdiego/compiledb#124]
On windows using a PATH variable for includes ends up with a string with no '\'
Example:
In Makefile:  -I$(VK_SDK_PATH)/include
When make is run: -IC:\VulkanSDK\1.3.216.0/include
In compile_commands.json:  -IC:VulkanSDK1.3.216.0/include

**Comment by arminveres** [comment]
I just noticed this issue as well. Is the PR #126 going to be merged i.e. the problem fixed in the foreseeable future?
Thanks for working on it @jinyi-x

**Comment by okuma10** [comment]
@arminveres did the fix worked for you? Nothing on my end

**Comment by arminveres** [comment]
> @arminveres did the fix worked for you? Nothing on my end

Hi, I didn't try. So I am guessing it didn't work for you?

**Comment by okuma10** [comment]
yes windows Path variables are without their '\' . I suppose it's a simple correction since everything is extracted correctly it just skips adding that character. I'll try to look into it next week.

**Comment by jinyi-x** [comment]
This patch fixed this problem on my system (Windows 10 LTSC 2021 & Python 3.10).And I noticed that if I switch my python into python3.6, it seems doesn't need any patch.Maybe it has relation with python version?I will keep testing.
来自 Outlook<http://aka.ms/weboutlook>
________________________________
发件人: Armin Veres ***@***.***>
发送时间: 2022年11月10日 20:38
收件人: nickdiego/compiledb ***@***.***>
抄送: jinyi-x ***@***.***>; Mention ***@***.***>
主题: Re: [nickdiego/compiledb] Windows PATH variables parsed with no '\' (Issue #124)


I just noticed this issue as well. Is the PR #126<https://github.com/nickdiego/compiledb/pull/126> going to be merged i.e. the problem fixed in the foreseeable future?
Thanks for working on it @jinyi-x<https://github.com/jinyi-x>

―

**Comment by okuma10** [comment]
So I looked into it today and it appears the issue is due to this module --> `bashlex`,  used to parse the command output from make 
![image](https://user-images.githubusercontent.com/44575116/201610157-3fc49587-3d82-45c0-9984-84d8044c6730.png)

as you can see the `preprocesed` variable has the correct string, but the output of the `bashlex.parser.parse()` function is without the `\\` character
![image](https://user-images.githubusercontent.com/44575116/201610902-b92cb46f-677a-4770-a584-8db1a20a336a.png)

**Comment by okuma10** [comment]
I think this issue is solved by the above comment, maybe author can apply better change, but I also found this, trying to solve why the vulkan header wasn't being detected by nvim's clangd lsp.
![image](https://user-images.githubusercontent.com/44575116/201855727-b721985b-351f-407e-87e1-33c25f2ac869.png)

Do you guy's still get the same `compile_commands.json` ...with the arguments having `" ' argument ' " ` instead of  `" argument "`? @jinyi-x @arminveres

**Comment by arminveres** [comment]
> I think this issue is solved by the above comment, maybe author can apply better change, but I also found this, trying to solve why the vulkan header wasn't being detected by nvim's clangd lsp. ![image](https://user-images.githubusercontent.com/44575116/201855727-b721985b-351f-407e-87e1-33c25f2ac869.png)
> 
> Do you guys still get the same `compile_commands.json` ...with the arguments having `" ' argument ' " ` instead of `" argument "`? @jinyi-x @arminveres

Nicely done. It really solves it. I am also gettting the " ' ' " quotation. I haven't investigated it, I only ran a quick sed from inside neovim and now it works!

I think you should propose a PR for this.

**Comment by okuma10** [comment]
@arminveres I looked up an old `compile_commands.json` generated from `compiledb`, so it might be something from the change. I'll take a look at it later and will try to find why it adds those quotation marks.

**Comment by okuma10** [comment]
@arminveres I looked it up and it appears the " ' ' " pattern is introduced by @jinyi-x 's solution of using `repr(s).encode().decode()` in the `unescape(s)` function at the end. 
Not sure what exactly that function does as there are no comments, but on my end Windows10 Python 3.11 returning to the original output of `s.encode().decode()` was giving me errors that cannot read a byte at the end of the string. So I just returned `s`, and that appeared to work. Might be a good idea for @nickdiego to give his input on this.
The `unescape()` function is used in the `parse_build_log()` function at around line 140(on my end since I might have commented out prints above it)
![image](https://user-images.githubusercontent.com/44575116/201969362-045eaaf4-6fe5-4873-8406-845943adffe6.png)

**Comment by arminveres** [comment]
@okuma10 It seems to work, if you drop jinyi's solution and just add yours. I have not tried it on Linux, though.

**Comment by GitMensch** [comment]
#128 seems to also take this into account...
